spec/services/work_packages/update_ancestors_service_spec.rb

```diff
@@ -795,6 +796,213 @@ def call_update_ancestors_service(work_package)
     end
   end
 
+  describe "simple average mode for total % complete calculation",
+           with_settings: { total_percent_complete_mode: "simple_average" } do
+    subject(:call_result) do
+      described_class.new(user:, work_package: parent)
+                      .call(%i(remaining_hours))
+    end
```

Comment:
That's not exactly how this class is supposed to be called: it is called on a child which has been modified, and we check that the ancestors got updated as well.

Calling it like you did actually hides a bug: in the `WorkPackages::UpdateAncestors::Loader`, there can be structured data (the `WorkPackageLikeStruct`), but also plain `WorkPackage` instances, and they not respond to `status_excluded_from_totals`.

I'll fix that.

Reply:
Fixed in 1874fb3003e.